src/components/widget/Checkbox.js

```diff
@@ -42,9 +53,9 @@ const Checkbox = props => {
         <input
           ref={rawWidget}
           type="checkbox"
-          checked={widgetData[0].value}
+          checked={checkedState}
           disabled={widgetData[0].readonly || disabled}
-          onChange={e => handlePatch(widgetField, e.target.checked, id)}
+          onChange={e => updateCheckedState(e)}
```

Comment:
why do we need an anonymous function here ?

Reply:
This function is used to do fine tuning to state before further calling the handlePatch. In our case set the state value checkedState L:34 .

Reply:
I still don't understand why there's an anonymous function there. If param set in onChange is just e and updateCheckedState expects e as well, why not just onChange={updateCheckedState} ?